File: Axiom/Internal/Ast/UnaryExpression.cs
namespace Axiom.Internal.Ast
{
    internal class UnaryExpression : Expression
    {
        public UnaryExpression(Token unaryOperator, Expression expression, bool isPostfix = false)
        {
            Operator = unaryOperator;
            Expression = expression;
            IsPostfix = isPostfix;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public Token Operator { get; private set; }
        public Expression Expression { get; private set; }
        public bool IsPostfix { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/UnaryExpression.java
package com.moybl.axiom.ast;

import com.moybl.axiom.Token;

public class UnaryExpression implements Expression {

	public enum Kind {
		POSTFIX, PREFIX
	}

	private Token operator;
	private Expression expression;
	private Kind kind;

	public UnaryExpression(Token operator, Expression expression, Kind kind) {
		this.operator = operator;
		this.expression = expression;
		this.kind = kind;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Token getOperator() {
		return operator;
	}

	public Expression getExpression() {
		return expression;
	}

	public Kind getKind() {
		return kind;
	}

}
